react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      counter: 0,
+      image: this.props.pokemon[0].url
+    };
+    this.next = this.next.bind(this);
+    this.prev = this.prev.bind(this);
+    this.openPokeball = this.openPokeball.bind(this);
+  }
+
+  componentDidMount() {
+    this.rotating = setInterval(this.next, 3000);
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.rotating);
+  }
+
+  next(e) {
+    if (this.state.counter < this.props.pokemon.length) {
+      this.setState({ counter: this.state.counter + 1 }, () => {
+        if (this.state.counter === this.props.pokemon.length) {
+          this.setState({ counter: 0 }, () => {
+            this.setState({ image: this.props.pokemon[this.state.counter].url });
+          });
+        } else {
+          this.setState({ image: this.props.pokemon[this.state.counter].url });
+        }
+      });
+    }
```

Comment:
Instead of setting state plus one and then undoing it, check what the correct next count should be.

Also, you don't need to store the image in state, since you are storing the index of the image.